iothub/service/tests/QueryResponseTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Azure.Devices.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class QueryResponseTests
+    {
+        [TestMethod]
+        public async Task QueryResponse_MoveNextAsync()
+        {
+            // arrange
+            string query = "select * from devices where deviceId = 'foo'";
+            var twin1 = new ClientTwin("foo");
+            var twin2 = new ClientTwin("bar");
```

Comment:
Nit: Either change the device Id of twin2 to "foo" or update L36 to return only twin 1